src/domain/accounts/domain/src/repos/access_token_repository.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use internal_error::InternalError;
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::{AccessToken, Account};
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait AccessTokenRepository: Send + Sync {
+    async fn create_access_token(
+        &self,
+        access_token: &AccessToken,
+    ) -> Result<(), CreateAccessTokenError>;
```

Comment:
Another confusing name. The token seems to be created already, since we pass it as a parameter. This is registering or saving the token the caller has already generated.